File: solutions/erlang/difference-of-squares/1/src/difference_of_squares.erl
-module(difference_of_squares).
-export([difference_of_squares/1, square_of_sum/1, sum_of_squares/1]).
-import(lists, [sum/1, map/2, seq/2]).

difference_of_squares(N) -> square_of_sum(N) - sum_of_squares(N).

square_of_sum(N) ->
  S = sum(seq(1,N)),
  S*S.

sum_of_squares(N) -> sum(map(fun(X)->X*X end, seq(1,N))).

File: solutions/lua/difference-of-squares/1/difference-of-squares.lua
local function square_of_sum(n)
    res = 0
    for i = 1, n do
        res = res + i
    end
    return res * res
end

local function sum_of_squares(n)
    res = 0
    for i = 1, n do
        res = res + (i * i)
    end
    return res
end

local function difference_of_squares(n)
    return square_of_sum(n) - sum_of_squares(n)
end

return { square_of_sum = square_of_sum, sum_of_squares = sum_of_squares, difference_of_squares = difference_of_squares }
